rpcclient/wallet.go

```diff
@@ -798,9 +798,14 @@ func (c *Client) PurchaseTicketAsync(fromAccount string,
 		expiryVal = *expiry
 	}
 
+	splittx := uint32(1)
+	if splitTx != nil {
+		splittx = uint32(*splitTx)
+	}
+
 	cmd := dcrjson.NewPurchaseTicketCmd(fromAccount, spendLimit.ToCoin(),
 		&minConfVal, &ticketAddrStr, &numTicketsVal, &poolAddrStr,
-		&poolFeesFloat, &expiryVal, nil, noSplitTransaction)
+		&poolFeesFloat, &expiryVal, dcrjson.String(""), &splittx)
```

Comment:
you can maintain passing `nil` for the comment param here since it's `*string`.

Reply:
somehow, passing nil there causes issues. The values of every field after the comment param won't be passed if nil is passed.

Reply:
It's a consequence of how the arguments are parsed.  This seems to be a common pitfall: https://github.com/decred/dcrrpcclient/pull/89

Reply:
Yes!! Because of the way rpcclient marshals parameters, every parameter after a nil param will default to it's zero value